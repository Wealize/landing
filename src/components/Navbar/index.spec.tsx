import React from 'react'
import { mount } from 'enzyme'

import ButtonRounded from '../Buttons/Rounded'

import NavBar from './index'

// eslint-disable-next-line
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
describe('<NavBar />', () => {
  useRouter.mockImplementation(() => ({
    pathname: '/'
  }))

  const componentProps = {
    showNavigationBarClosablePage: true
  }
  const wrapperWithClosablePage = mount(<NavBar {...componentProps} />)

  it('should render', () => {
    expect(wrapperWithClosablePage).toBeDefined()
  })

  it('should render content correctly', () => {
    expect(wrapperWithClosablePage.find(ButtonRounded).length).toEqual(1)

    const componentProps = {
      showNavigationBarClosablePage: false
    }
    const wrapperWithoutClosablePage = mount(<NavBar {...componentProps} />)
    expect(wrapperWithoutClosablePage.find(ButtonRounded).length).toEqual(0)
  })
})
