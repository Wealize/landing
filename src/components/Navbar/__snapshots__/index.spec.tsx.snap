// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render 1`] = `
<Memo(NavBar)
  showNavigationBarClosablePage={true}
>
  <styled.nav
    id="navbar"
    isVisible={true}
    showNavigationBarClosablePage={true}
  >
    <nav
      className="hxNwcS"
      id="navbar"
    >
      <styled.div
        data-cy="navbar-logo-mobile-container"
      >
        <div
          className="jcroSH"
          data-cy="navbar-logo-mobile-container"
        >
          <Link
            href="/"
          >
            <a
              aria-label="Home"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <Memo(WealizeMinimalMediumIcon)>
                <svg
                  height="3em"
                  viewBox="0 0 45 24.824"
                  width="3em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.612 24.824L4.737 6.473H-.001V.001h10.472a.328.328 0 01.315.236l4.178 14.453L19.981.001h8.217l4.947 14.741L37.393.001h7.6l-7.867 24.824h-7.708L23.893 8.794l-5.419 16.031z"
                    fill="#f06078"
                  />
                </svg>
              </Memo(WealizeMinimalMediumIcon)>
            </a>
          </Link>
        </div>
      </styled.div>
      <styled.div
        data-cy="navbar-logo-container"
      >
        <div
          className="iXyhlw"
          data-cy="navbar-logo-container"
        >
          <Link
            href="/"
          >
            <a
              aria-label="Home"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <Memo(WealizeNormalMediumIcon)>
                <svg
                  height="100%"
                  viewBox="0 0 130 27.463"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M111.564 17.848a9.181 9.181 0 018.874-9.5c.1-.006.216-.01.33-.006 5.713 0 9.229 3.443 9.229 9.251a10.934 10.934 0 01-.114 1.7h-12.835a3.97 3.97 0 004.2 3.482 3.565 3.565 0 003.707-2.358l4.71 1.3c-.889 3.12-3.821 5.735-8.506 5.735a9.286 9.286 0 01-9.595-9.604zm5.6-2.1h7.389c-.089-1.381-1.041-3.047-3.72-3.047a3.487 3.487 0 00-3.67 3.041zm-83.851 2.1a9.181 9.181 0 018.874-9.5c.1-.006.2-.01.317-.006 5.713 0 9.242 3.443 9.242 9.251a12.741 12.741 0 01-.114 1.7H38.797a3.957 3.957 0 004.189 3.482 3.555 3.555 0 003.707-2.358l4.71 1.3c-.889 3.12-3.821 5.735-8.493 5.735a9.286 9.286 0 01-9.598-9.604zm5.586-2.1h7.389c-.076-1.381-1.041-3.047-3.707-3.047a3.5 3.5 0 00-3.683 3.041zm14.993 6.2c0-3.19 2.412-5 5.446-5.44l4.342-.654c1-.143 1.333-.616 1.333-1.231 0-1.054-.927-1.961-2.7-1.961a2.964 2.964 0 00-3.161 2.647l-4.887-.978c.228-2.577 2.666-5.986 8.087-5.986 5.967 0 8.163 3.266 8.163 6.963v8.816a17.181 17.181 0 00.216 2.793h-5.078a9.1 9.1 0 01-.19-2.034 5.848 5.848 0 01-5.192 2.542c-4.037 0-6.373-2.609-6.373-5.478zm7.642-2.355c-1.193.181-2.082.724-2.082 1.993 0 .946.635 1.85 2.2 1.85 1.663 0 3.364-.8 3.364-3.593v-.8zm32.626 7.325v-4.859l8.125-8.452h-7.9V8.891h15.095v4.535l-8.493 8.74h8.645v4.751zm-9.038 0V8.887h5.636v18.026zm-77.416 0L3.44 13.592H0v-4.7h7.6a.238.238 0 01.229.171l3.034 10.5 3.644-10.667h5.967l3.593 10.7 3.085-10.7h5.522l-5.713 18.026h-5.6l-4.012-11.641-3.936 11.641zm67.17 0V.648h5.65V26.91zm9.763-23.5V3.38a3.3 3.3 0 113.376 3.231h-.076a3.266 3.266 0 01-3.3-3.193z"
                    fill="#f06078"
                  />
                </svg>
              </Memo(WealizeNormalMediumIcon)>
            </a>
          </Link>
        </div>
      </styled.div>
      <Memo(ButtonRounded)
        animatedBackgroundColor="#6666ff"
        animatedTextColor="#ffffff"
        aria-label="Go to home page"
        backgroundColor="#ffffff"
        handleClick={[Function]}
        size={
          Object {
            "desktop": Array [
              170,
              50,
            ],
            "mobile": Array [
              90,
              50,
            ],
            "tablet": Array [
              150,
              50,
            ],
            "wide": Array [
              190,
              50,
            ],
          }
        }
        textColor="#6666ff"
      >
        <styled.div
          animatedBackgroundColor="#6666ff"
          animatedTextColor="#ffffff"
          backgroundColor="#ffffff"
          className="button-rounded"
          onClick={[Function]}
          size={
            Object {
              "desktop": Array [
                170,
                50,
              ],
              "mobile": Array [
                90,
                50,
              ],
              "tablet": Array [
                150,
                50,
              ],
              "wide": Array [
                190,
                50,
              ],
            }
          }
          textColor="#6666ff"
        >
          <div
            className="hmDnaE button-rounded"
            onClick={[Function]}
            size={
              Object {
                "desktop": Array [
                  170,
                  50,
                ],
                "mobile": Array [
                  90,
                  50,
                ],
                "tablet": Array [
                  150,
                  50,
                ],
                "wide": Array [
                  190,
                  50,
                ],
              }
            }
          >
            <styled.button
              aria-label="Return"
              backgroundColor="#ffffff"
            >
              <button
                aria-label="Return"
                className="dbnkaG"
              >
                <styled.span
                  textColor="#6666ff"
                >
                  <span
                    className="hEOWLJ"
                  >
                    Return
                  </span>
                </styled.span>
              </button>
            </styled.button>
            <styled.button
              animatedBackgroundColor="#6666ff"
              aria-label="Return animated"
            >
              <button
                aria-label="Return animated"
                className="eobazg"
              >
                <styled.span
                  textColor="#6666ff"
                >
                  <span
                    className="hEOWLJ"
                  >
                    Return
                  </span>
                </styled.span>
              </button>
            </styled.button>
          </div>
        </styled.div>
      </Memo(ButtonRounded)>
    </nav>
  </styled.nav>
</Memo(NavBar)>
`;
